extensions/azure/assetindex-cosmos/src/main/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetIndex.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright (c) 2020, 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.assetindex.azure;
+
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.NotFoundException;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.PartitionKey;
+import net.jodah.failsafe.RetryPolicy;
+import org.eclipse.dataspaceconnector.assetindex.azure.model.AssetDocument;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.asset.AssetIndex;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static net.jodah.failsafe.Failsafe.with;
+
+public class CosmosAssetIndex implements AssetIndex {
+
+    private final CosmosContainer container;
+    private final CosmosQueryRequestOptions tracingOptions;
+    private final TypeManager typeManager;
+    private final String partitionKey;
+    private final RetryPolicy<Object> retryPolicy;
+
+    /**
+     * Creates a new instance of the CosmosDB-based for Asset storage.
+     *
+     * @param container   The CosmosDB container.
+     * @param typeManager The {@link TypeManager} that's used for serialization and deserialization
+     */
+    public CosmosAssetIndex(CosmosContainer container, String partitionKey, TypeManager typeManager, RetryPolicy<Object> retryPolicy) {
+        this.container = container;
+        this.typeManager = typeManager;
+        this.partitionKey = partitionKey;
+        tracingOptions = new CosmosQueryRequestOptions();
+        tracingOptions.setQueryMetricsEnabled(true);
```

Comment:
If this is a configuration set from within the EDC it would be nice to enable/disable this via EDC configuration setting.

Reply:
Indeed you are right, I added this as a parameter in the AbstractCosmosConfig class